pkg/config/kubeClient.go

```diff
@@ -53,17 +69,20 @@ func RESTClient(cfg *rest.Config) *rest.RESTClient {
 	return restClient
 }
 
-func GetConfig() *rest.Config {
-	var clientConfig *rest.Config
-	if Cfg.KubeConfig != "" {
-		glog.V(2).Infof("Creating k8s client using path: %s", Cfg.KubeConfig)
-		clientConfig, _ = clientcmd.BuildConfigFromFlags("", Cfg.KubeConfig)
-	} else {
-		glog.V(2).Info("Creating k8s client using InClusterlientConfig()")
-		clientConfig, _ = rest.InClusterConfig()
-	}
-	return clientConfig
+// Get the kubernetes dynamic client.
+func GetDynamicClient() dynamic.Interface {
+    mutex.Lock()
+    defer mutex.Unlock()
+    if dynamicClient != nil {
+        return dynamicClient
+    }
+    newDynamicClient, err := dynamic.NewForConfig(GetConfig())
+    if err != nil {
+        glog.Fatal("Cannot Construct Dynamic Client ", err)
```

Comment:
Curious if we have to return nil .. please check

Reply:
Please ignore , we call Fatal .

Reply:
Using Fatal here - dont need a return